app/uk/gov/hmrc/individualsincomeapi/domain/integrationframework/paye/IFPayeEntry.scala

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualsincomeapi.domain.integrationframework.paye
+
+import play.api.libs.functional.syntax.{unlift, _}
+import play.api.libs.json.{Format, JsPath, Reads}
+import play.api.libs.json.Reads.{maxLength, minLength, pattern, verifying}
+import uk.gov.hmrc.individualsincomeapi.domain.integrationframework.paye.IFPaye.paymentAmountValidator
+
+case class IFGrossEarningsForNics(
+                                   inPayPeriod1: Option[Double],
+                                   inPayPeriod2: Option[Double],
+                                   inPayPeriod3: Option[Double],
+                                   inPayPeriod4: Option[Double]
+                                 )
+
+case class IFTotalEmployerNics(
+                           inPayPeriod1: Option[Double],
+                           inPayPeriod2: Option[Double],
+                           inPayPeriod3: Option[Double],
+                           inPayPeriod4: Option[Double],
+                           ytd1: Option[Double],
+                           ytd2: Option[Double],
+                           ytd3: Option[Double],
+                           ytd4: Option[Double]
+                         )
+
+case class IFEmployeeNics(
+                           inPayPeriod1: Option[Double],
+                           inPayPeriod2: Option[Double],
+                           inPayPeriod3: Option[Double],
+                           inPayPeriod4: Option[Double],
+                           ytd1: Option[Double],
+                           ytd2: Option[Double],
+                           ytd3: Option[Double],
+                           ytd4: Option[Double]
+                         )
+
+case class IFEmployeePensionContribs(
```

Comment:
I think the convention for initialisms in class names is still camel case (i.e. IfEmployeePensionContribs) https://docs.scala-lang.org/style/naming-conventions.html